service/history/historyEngine.go

```diff
@@ -696,12 +696,18 @@ Update_History_Loop:
 					failCause = workflow.DecisionTaskFailedCause_BAD_START_TIMER_ATTRIBUTES
 					break Process_Decision_Loop
 				}
+				if !userTimersLoaded {
```

Comment:
You likely need a GetTimers() method that lazy loads and returns all the timers for timerProcessor (instead of LoadUserTimers). Your callers just need to add / remove or read all the timers, loading (only once) is an internal impl detail IMO. Will leave it up to you.